core/primitives-core/src/account.rs

```diff
@@ -154,6 +191,29 @@ impl Account {
         }
     }
 
+    #[inline]
+    pub fn global_contract_hash(&self) -> Option<CryptoHash> {
+        match self {
+            Self::V1(_) => None,
+            Self::V2(account) => match account.account_contract.clone() {
+                AccountContractType::GlobalContract(hash)
+                | AccountContractType::GlobalContractByAccount(_, hash) => Some(hash),
+                _ => None,
+            },
+        }
+    }
+
+    #[inline]
+    pub fn global_contract_account_id(&self) -> Option<AccountId> {
```

Comment:
avoid cloning `AccountId`, make return type `Option<&AccountId>` and make it clients responsibility to call `.cloned()` if needed